tests/unittests/Foundation/NSDictionaryTests.mm

```diff
@@ -68,3 +68,51 @@
 
     ASSERT_EQ(expectedCount, [dictionary count]);
 }
+
+TEST(NSDictionary, KeysSortedByValue) {
+    NSDictionary* dictionary = @{@"f" : @6, @"b": @2, @"a" : @1, @"c" : @3, @"e" : @5, @"d" : @4};
+    NSArray* expected = @[@"a", @"b", @"c", @"d", @"e", @"f"];
+    NSArray* actual = [dictionary keysSortedByValueUsingComparator:^NSComparisonResult(id obj1, id obj2) {
+        int a = [obj1 intValue];
+        int b = [obj2 intValue];
+        if (a == b) {
+            return NSOrderedSame;
+        }
+
+        return (a > b) ? NSOrderedDescending : NSOrderedAscending;
+    }];
+
+    ASSERT_OBJCEQ(expected, actual);
+}
+
+TEST(NSDictionary, KeysSortedByValueWithOptions) {
+    NSDictionary* dictionary = @{@"f" : @6, @"b": @2, @"a" : @1, @"c" : @3, @"e" : @5, @"d" : @4};
+    NSArray* expected = @[@"a", @"b", @"c", @"d", @"e", @"f"];
+    NSArray* actual = [dictionary keysSortedByValueWithOptions:0 usingComparator:^NSComparisonResult(id obj1, id obj2) {
+        int a = [obj1 intValue];
+        int b = [obj2 intValue];
+        if (a == b) {
+            return NSOrderedSame;
+        }
+
+        return (a > b) ? NSOrderedDescending : NSOrderedAscending;
+    }];
+
+    ASSERT_OBJCEQ(expected, actual);
+}
+
+TEST(NSDictionary, KeysSortedByValueWithOptions_Stable) {
+    NSDictionary* dictionary = @{@"a" : @1, @"b": @1, @"c" : @1, @"d" : @1, @"e" : @1, @"f" : @0};
+    NSArray* expected = @[@"f", @"d", @"b", @"e", @"c", @"a"]; // Note: ordering after "f" is dependent on CFDictionary (this ordering matches the reference platform)
+    NSArray* actual = [dictionary keysSortedByValueWithOptions:NSSortStable usingComparator:^NSComparisonResult(id obj1, id obj2) {
```

Comment:
It might be better to test NSArray's sort directly -- since it doesn't make sense to force a stable sort on the keys when the keys are not in a known order.

Something like this: Sort `[3, 100, 2, 200, 1, 300]`  with a comparator that says 1-3 and 100-300 are the "same". Then check for `[3, 2, 1, 100, 200, 300]`